diffusion/models/stable_diffusion.py

```diff
@@ -448,14 +452,24 @@ def generate(
         added_cond_kwargs = {}
         # if using SDXL, prepare added time ids & embeddings
         if self.sdxl and pooled_embeddings is not None:
-            if not crop_params:
-                crop_params = [0., 0.]
-            if not size_params:
-                size_params = [width, height]
-            add_time_ids = torch.tensor([[width, height, *crop_params, *size_params]], dtype=torch.float, device=device)
-            add_time_ids = add_time_ids.repeat(pooled_embeddings.shape[0], 1)
-            add_text_embeds = pooled_embeddings
+            if crop_params is None:
+                crop_params = torch.tensor([0., 0.]).repeat(batch_size, 1)
+            elif isinstance(crop_params, tuple):  # convert to tensor
+                crop_params = torch.tensor([crop_params[0], crop_params[1]], dtype=torch.float32).repeat(batch_size, 1)
+            if input_size_params is None:
+                input_size_params = torch.tensor([width, height], dtype=torch.float32).repeat(batch_size, 1)
+            elif isinstance(input_size_params, tuple):  # convert to tensor
+                input_size_params = torch.tensor([input_size_params[0], input_size_params[1]],
+                                                 dtype=torch.float32).repeat(batch_size, 1)
+            output_size_params = torch.tensor([width, height], dtype=torch.float32).repeat(batch_size, 1)
 
+            if do_classifier_free_guidance:
+                crop_params = torch.cat([crop_params, crop_params])
+                input_size_params = torch.cat([input_size_params, input_size_params])
+                output_size_params = torch.cat([output_size_params, output_size_params])
+
+            add_time_ids = torch.cat([input_size_params, crop_params, output_size_params], dim=1).to(device)
+            add_text_embeds = pooled_embeddings
             added_cond_kwargs = {'text_embeds': add_text_embeds, 'time_ids': add_time_ids}
```

Comment:
Any reason to not do this?
```suggestion
            added_cond_kwargs = {'text_embeds': pooled_embeddings, 'time_ids': add_time_ids}
```